SharpPcap/LibPcap/PcapDevice.cs

```diff
@@ -586,6 +586,51 @@ public virtual void SendPacket(byte[] p, int size)
         /// <param name="size">The number of bytes to send</param>
         public abstract void SendPacket(ReadOnlySpan<byte> p);
 
+        #region Timestamp
+        public virtual TimestampResolution TimestampPrecision
+        {
+            get => (TimestampResolution)LibPcapSafeNativeMethods.pcap_get_tstamp_precision(PcapHandle);
+            set => LibPcapSafeNativeMethods.pcap_set_tstamp_precision(PcapHandle, (int)value);
+        }
+
+        public static string TimestampName(TimestampType number)
+        {
+            var ptr =  LibPcapSafeNativeMethods.pcap_tstamp_type_val_to_name((int)number);
+            return Marshal.PtrToStringAnsi(ptr);
+
+        }
+
+        public static string TimestampDescription(TimestampType number)
+        {
+            var ptr = LibPcapSafeNativeMethods.pcap_tstamp_type_val_to_description((int)number);
+            return Marshal.PtrToStringAnsi(ptr);
+        }
+
+        public List<TimestampType> TimestampsSupported
+        {
+            get
+            {
+                IntPtr typePtr = IntPtr.Zero;
+
+                // Note: typePtr must be freed with pcap_free_tstamp_types()
+                var typeCount = LibPcapSafeNativeMethods.pcap_list_tstamp_types(PcapHandle, ref typePtr);
```

Comment:
@chmorgan if `typeCount == 0`, return a list containing only the element `PCAP_TSTAMP_HOST`
See https://www.tcpdump.org/manpages/pcap_list_tstamp_types.3pcap.html

Reply:
good catch!